### ISSUE TITLE
SQL issues with setup

### ISSUE THREAD

**Issue by camjac251** [TheReverend403/uPste#2]
I'm trying to set this up on Ubuntu 14.04 with mariadb (drop in replacement for mysql) and haven't had much luck. I changed the info in .env and tried the artisan migrate command but it gave me this error
`[PDOException]                                                                           
  SQLSTATE[28000] [1045] Access denied for user 'upste'@'localhost' (using password: YES)`

I then tried configuring it with the default sqlite settings but it gave me this error
`[InvalidArgumentException]                                                      
  Database (/usr/share/nginx/html/upste/storage/database.sqlite) does not exist`

I did `touch database.sqlite` in the storage folder and it finally gave this

`Migration table created successfully.
 [Doctrine\DBAL\DBALException]                                                                 
  Operation 'Doctrine\DBAL\Platforms\AbstractPlatform::getAlterTableSQL' is not supported by p  
  latform.`

I am not sure how to continue. I'm using PHP 5.5.34-1, MariaDB 10.1.13 (mysql Ver 15.1), nginx 1.9.13, Latest Composer dev version as of today, sqlite 3.8.2, and finally NodeJS version v5.10.0
and NPM 3.8.3.

**Comment by lwatsondev** [comment]
It seems like your `upste` user isn't configured to be allowed connect from `localhost`, and mysql is really funny about that, for example, 127.0.0.1 is not the same as localhost.

Post the output of `SELECT User,Host FROM mysql.user;`, omitting anything sensitive.

**Comment by camjac251** [comment]
Here it is. It's not outputting the host though.

> MariaDB [(none)]> SELECT User FROM mysql.user;
> +------------------+
> | User             |
> +------------------+
> | root             |
> | root             |
> | debian-sys-maint |
> | root             |
> | upste            |
> +------------------+
> 5 rows in set (0.00 sec)

**Comment by lwatsondev** [comment]
Yeah, my bad. `SELECT User,Host FROM mysql.user;`

**Comment by camjac251** [comment]
Here's the results
![xshell_2016-04-01_19-55-17](https://cloud.githubusercontent.com/assets/6313132/14223347/ac019a12-f843-11e5-949f-5c43ff80bfed.png)

**Comment by lwatsondev** [comment]
Interesting. Can you connect manually with `mysql -u upste -p -h localhost` using the password copied directly out of .env? You can use Shift + Insert to paste into most Linux terminal emulators.

**Comment by camjac251** [comment]
I was able to connect using the password from the .env. I was assuming that it was not on the default port that it was searching for. but sadly the script still doesnt connect

**Comment by lwatsondev** [comment]
Also just to make sure, did you run something along the lines of `GRANT ALL ON upste.* to upste@localhost IDENTIFIED BY 'password';` to create the upste user?

**Comment by camjac251** [comment]
For some reason it works now. That was really weird. Not sure what went wrong there as I had tried using the root mysql user at one point with the same error.

**Comment by lwatsondev** [comment]
What did you change?

**Comment by camjac251** [comment]
Not sure. before creating the issue, I had changed it back to sqlite and was trying that out. I just did the command you said last and changed the .env to mysql again and ran the artisan migrate, somehow working that time.

**Comment by lwatsondev** [comment]
Ah, I see. It seems you never actually gave the upste user any access to the database. Glad you got it solved anyway.

**Comment by camjac251** [comment]
In trying to install the script again I think I had found out my initial mistake that I once again had today.

It could either be one of these things;

When I had first installed I did `GRANT ALL PRIVILEGES ON upste.* TO upstedb@localhost;` instead of `GRANT ALL ON upstedb.* to upste@localhost IDENTIFIED BY 'password';` Although I did try the second command after the first with no change in the error message so it may not be the problem.
or
It could be that when I first tried it, I had capital letters in the password and it would only take it without any. My mysql password currently only has lowercase numbers and letters, they both had the same amount of characters.
